### PR TITLE
create playlist plugin

### PR DESCRIPTION
**Pull request by entonbiba**
### create playlist plugin
  - user util.ajax to get file content - **ok**
  - support m3u - **ok**
  - check if mp3 or wav file urls set in playlist - **ok**
  - return empty array if playlist type not supported or set - **ok**
  - create plugin example and docs - **ok** #1019 

### Parameters:
**playlistFile** - link to playlist file

**playlistType**
'm3u' or 'audio/mpegurl' will default to m3u

Preview:
http://codepen.io/entonbiba/pen/OpMQjR

### CODE REVIEW

**Comment by entonbiba** [comment]
@thijstriemstra yes don't merge until I complete the remaining tasks, I added the label 'needs-update'.

**Comment by thijstriemstra** [comment]
@entonbiba i'm not sure if thats the way to go. maybe you should not make a PR before that checklist is completed?

**Comment by mspae** [comment]
I don't have any strong opinions about this. I think we all agree the PR should only be merged when it's ready. But whether PRs should only be opened when they are truly ready (or the author considers them to be ready at least) I don't know – It does add clutter to the PR list and may be confusing if everybody starts doing it.

**Comment by katspaugh** [comment]
I think it's OK to open a PR to discuss the approach before it's too late. Also, is `m3u` the most popular playlist format? If so, this plugin would already be useful to people even without other formats.

**Comment by entonbiba** [comment]
@mspae @thijstriemstra @katspaugh yea I opened a pr so I can share the code updates directly here instead of linking to them.

I'll remove the other formats for now and create the example for m3u.

**Comment by entonbiba** [comment]
@thijstriemstra @katspaugh the playlist parser is now ready 👍

**Comment by entonbiba** [comment]
@thijstriemstra sure, all agree on this @katspaugh @mspae ? `PlayList`

**Comment by mspae** [comment]
@entonbiba I like that name.

File: plugin/wavesurfer.playlistparser.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        } else {
+            throw new Error('No playlist file provided');
+        }
+    },
+
+    parsedList: function() {
+        return this.playlistData;
+    },
+
+    parse: function(playlistFile) {
```
Comment (thijstriemstra) [inline plugin/wavesurfer.playlistparser.js]: parse sounds too techy, I'd personally call it load. And I also prefer `playlistData` over `parsedList`, can you come up with a better name that avoids `parse`?

File: plugin/wavesurfer.playlistparser.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        // check if playlist type is given
+        var playlist = [];
+
+        if (this.playlistType == 'm3u' || this.playlistType == 'audio/mpegurl') {
+            playlist = playlistFile.replace(/^.*#.*$|#EXTM3U|#EXTINF:/mg, '').split('\n');
+        } else {
+            throw new Error('No valid playlist file provided, valid formats are m3u pls smil json or their valid mime types');
```
Comment (thijstriemstra) [inline plugin/wavesurfer.playlistparser.js]: I would remove the listing of the supported formats in the error and just keep it short: No valid playlist file provided.

**Comment by entonbiba** [comment]
@thijstriemstra updated, I changed it to getPlaylist and loadPlaylist

**Pull request merged**